redaxo/src/addons/media_manager/pages/types.php

```diff
@@ -44,6 +44,23 @@
     $func = '';
 }
 
+//-------------- copy type
+if ($func == 'copy' && $type_id > 0) {
+    $sql = rex_sql::factory();
+    
+    try {
+        $sql->setQuery('INSERT INTO '.rex::getTablePrefix() . 'media_manager_type (status, name, description) SELECT status, CONCAT(name, \' '.rex_i18n::msg('media_manager_type_name_copy').'\'), description FROM '.rex::getTablePrefix() . 'media_manager_type WHERE id = ?', [$type_id]);
+        $newTypeId = $sql->getLastId();
+        $sql->setQuery('INSERT INTO '.rex::getTablePrefix() . 'media_manager_type_effect (type_id, effect, parameters, priority, updatedate, updateuser, createdate, createuser) SELECT ?, effect, parameters, priority, updatedate, updateuser, createdate, createuser FROM '.rex::getTablePrefix() . 'media_manager_type_effect WHERE type_id = ?', [$newTypeId, $type_id]);
```

Comment:
@gharlan hier bin ich noch unschlüssig ob man hier nicht ggf. besser das ganze mit "richtigen rex_sql objekten" löst, anstatt alles in purem sql zu machen. der code wie er ist, ist aktuell sehr "fehleranfällig" insofern neue db spalten dazukommen oder vorhandene umbenannt werden sollten.

habs aber noch nicht zu ende gedacht wie es aussehen könnte.